modules/openapi-generator/src/main/resources/Java/libraries/retrofit2/JSON_jackson.mustache

```diff
@@ -30,7 +30,7 @@ public class JSON implements ContextResolver<ObjectMapper> {
     mapper = JsonMapper.builder()
             .serializationInclusion(JsonInclude.Include.NON_NULL)
             .configure(MapperFeature.ALLOW_COERCION_OF_SCALARS, false)
-            .configure(DeserializationFeature.FAIL_ON_UNKNOWN_PROPERTIES, true)
+            .configure(DeserializationFeature.FAIL_ON_UNKNOWN_PROPERTIES, {{failOnUnknownProperties}})
```

Comment:
Hi, yes, I can do this, the only issue is that some libraries are configured to fail on unknown properties, some aren't.

I see that `apache-httpclient`, `feign`, `google-api-client`, `restassured`, `restclient`, `resteasy`, `vertx` and `webclient` currently have `DeserializationFeature.FAIL_ON_UNKNOWN_PROPERTIES` set to `false`, whereas `jersey2`, `jersey3`, `native` and `retrofit2` have `DeserializationFeature.FAIL_ON_UNKNOWN_PROPERTIES` set to `true` 

Is there an elegant way to tell JavaClientCodegen to default the `failOnUnknownProperties` variable to true or false, depending of the chosen library?

Reply:
Good point.

My preferred approach would be to use this opportunity to unify the behaviour of the different libraries. Since the majority uses `false` as default, let's make `false` the default value of `failOnUnknownProperties` for all libraries.
This would also make this PR a "breaking change with fallback", i.e. it should become part of OpenAPI Generator 7.8.0, not of 7.7.1. But I think wing328 will take care of this aspect.

Reply:
It's done!

Reply:
Thank you!

Some tests in `samples/openapi3/client/petstore/java/jersey2-java8/` are failing now. Apparently due to the custom deserialization logic in this sample, which relies on `FAIL_ON_UNKNOWN_PROPERTIES` being `true`. Can you please fix this by simply adding in `bin/configs/java-jersey2-8.yaml` the following line somewhere under `additionalProperties:`?
```java
  failOnUnknownProperties: true
```

Reply:
Fixed! Thanks for the hint, I had trouble finding where it came from